functions.php

```diff
@@ -1385,90 +1385,91 @@ function wp_autoupdates_disable_auto_updates() {
 
 	// Check Asset Type.
 	if ( 'plugin' === $type ) {
-		$wp_autoupdate_plugins = get_site_option( 'wp_auto_update_plugins', array() );
-		if ( is_array( $wp_autoupdate_plugins ) && ! empty( $wp_autoupdate_plugins ) ) {
-			foreach ( $wp_autoupdate_plugins as $index => $plugin_file ) {
-				if ( $plugin_file === $asset ) {
-					unset( $wp_autoupdate_plugins[ $index ] );
-					update_site_option( 'wp_auto_update_plugins', $wp_autoupdate_plugins );
-
-					/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
-					$all_plugins           = apply_filters( 'all_plugins', get_plugins() );
-					$wp_autoupdate_plugins = array_intersect( $wp_autoupdate_plugins, array_keys( $all_plugins ) );
-					$enabled_count         = count( $wp_autoupdate_plugins );
-					$plugin_name           = $all_plugins[ $asset ]['Name'];
-
-					$plugin_url = add_query_arg(
-						array(
-							'action'   => 'autoupdate',
-							'plugin'   => $asset,
-							'_wpnonce' => wp_create_nonce( 'autoupdate-plugin_' . $asset ),
-						),
-						'plugins.php'
-					);
-
-					$return_html = sprintf(
-						'<p class="plugin-autoupdate-disabled"><a href="%s" class="edit plugin-autoupdate-enable" aria-label="%s"><span class="plugin-autoupdate-label">%s</span></a></p>',
-						esc_url_raw( $plugin_url ),
-						esc_html__( 'Auto-updates enabled', 'wp-autoupdate' ),
-						esc_html__( 'Enable auto-updates', 'wp-autoupdates' )
-					);
-
-					wp_send_json_success(
-						array(
-							'enabled_count'  => '(' . absint( $enabled_count ) . ')',
-							'disabled_count' => '(' . absint( count( $all_plugins ) - $enabled_count ) . ')',
-							'return_html'    => wp_kses_post( $return_html ),
-							'type'           => 'plugin',
-						)
-					);
-					break;
-				}
-			}
-		}
+		$wp_autoupdate_plugins = (array) get_site_option( 'wp_auto_update_plugins', array() );
+		$wp_autoupdate_plugins = array_diff( $wp_autoupdate_plugins, array( $asset ) );
+		update_site_option( 'wp_auto_update_plugins', $wp_autoupdate_plugins );
+
+		/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+		$all_plugins           = apply_filters( 'all_plugins', get_plugins() );
+		$wp_autoupdate_plugins = array_intersect( $wp_autoupdate_plugins, array_keys( $all_plugins ) );
+		$enabled_count         = count( $wp_autoupdate_plugins );
```

Comment:
Is there a reason why this should be recalculated every time the autoupdate setting for a single plugin is enabled or disabled?